modules/network/vpc/main.tf

```diff
@@ -33,28 +33,47 @@ locals {
     role                  = null
   }
 
-  primary_subnet = var.primary_subnetwork == null ? local.default_primary_subnet : var.primary_subnetwork
+  # Identify user-supplied primary subnetwork
+  # (1) explicit var.primary_subnetwork
+  # (2) explicit var.subnetworks[0]
+  # (3) implicit local default subnetwork
+  input_primary_subnetwork = try(coalesce(
+    var.primary_subnetwork,
+    try(var.subnetworks[0], null)
+  ), local.default_primary_subnetwork)
+
+  # Identify user-supplied additional subnetworks
+  # (1) explicit var.additional_subnetworks
+  # (2) explicit var.subnetworks[1:end]
+  # (3) empty list
+  input_additional_subnetworks = try(coalescelist(
+    var.additional_subnetworks,
+    try(slice(var.subnetworks, 1, length(var.subnetworks)), []),
+  ), [])
+
+  # at this point we have constructed a list of subnetworks but need to extract
+  # user-provided CIDR blocks or calculate them from user-provided new_bits
+  # after we complete deprecation, local.all_subnetworks can be replaced with
+  # var.subnetworks (or local.default_primary_subnetwork if that is null)
+  input_subnetworks = concat([local.input_primary_subnetwork], local.input_additional_subnetworks)
```

Comment:
Is there a way of verifying that input_primary_subnetwork and input_additional_subnetworks are using the same option for "new_bits" vs "subnet_ip"?

Reply:
`validation` blocks only support acting upon the `variable` they are inside. Which, to my knowledge, is the only way to validate inputs in a way that generates a useful error message for the user. Broadly, the goal here is to improve the experience for new variables while retaining the same level of functionality for existing variables.

That said, the old variables only support `new_bits` and we _can_ separately impose that each of them only has `new_bits` which would satisfy the desire here.

Reply:
I was hesitant to add validation to the deprecated variable because none was there before, but I've added it on HEAD of PR.

Reply:
The validation in that case would probably save some confusion, though I agree with the goal of keeping the deprecated variables as minimal as possible for now. Thanks for the context!